File: packages/sdk-dotnet/src/Notifique/Models/WhatsApp/WhatsAppInstanceQrResponse.cs
using System.Text.Json.Serialization;

namespace Notifique.Models.WhatsApp;

public record WhatsAppInstanceQrResponse(
    [property: JsonPropertyName("success")] bool Success,
    [property: JsonPropertyName("data")] WhatsAppInstanceQrData Data
);

public record WhatsAppInstanceQrData(
    [property: JsonPropertyName("status")] string Status,
    [property: JsonPropertyName("base64")] string? Base64
);

File: packages/sdk-java/src/main/java/com/notifique/sdk/model/WhatsAppInstanceQrResponse.java
package com.notifique.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class WhatsAppInstanceQrResponse {
    private boolean success;
    private Map<String, Object> data; // status, base64

    public boolean isSuccess() { return success; }
    public void setSuccess(boolean success) { this.success = success; }
    public Map<String, Object> getData() { return data; }
    public void setData(Map<String, Object> data) { this.data = data; }
}
